src/main/java/com/conveyal/r5/analyst/TravelTimeSurfaceComputer.java

```diff
@@ -102,22 +104,71 @@ public void write (OutputStream os) throws IOException {
 
             // Perform street search to find transit stops and non-transit times.
             StreetRouter sr = new StreetRouter(network.streetLayer);
-            sr.streetMode = accessMode;
             sr.profileRequest = request.request;
-            sr.distanceLimitMeters = 2000; // TODO hardwired same as gridcomputer
-            sr.setOrigin(request.request.fromLat, request.request.fromLon);
-            sr.dominanceVariable = StreetRouter.State.RoutingVariable.DISTANCE_MILLIMETERS;
-            sr.route();
 
-            // Get the travel times to all stops reached in the initial on-street search. Convert distances to speeds.
+            TIntIntMap accessTimes;
             int offstreetTravelSpeedMillimetersPerSecond = (int) (request.request.getSpeed(accessMode) * 1000);
+            int[] nonTransitTravelTimesToDestinations;
+
+            if (request.request.accessModes.contains(LegMode.CAR_PARK)) {
+                //Currently first search from origin to P+R is hardcoded as time dominance variable for Max car time seconds
+                //Second search from P+R to stops is not actually a search we just return list of all reached stops for each found P+R.
+                // If multiple P+Rs reach the same stop, only one with shortest time is returned. Stops were searched for during graph building phase.
+                // time to stop is time from CAR streetrouter to stop + CAR PARK time + time to walk to stop based on request walk speed
+                //by default 20 CAR PARKS are found it can be changed with sr.maxVertices variable
+                sr = PointToPointQuery.findParkRidePath(request.request, sr, network.transitLayer);
+
+                if (sr == null) {
+                    // Origin not found. Return an empty access times map, as is done by the other conditions for other modes.
+                    // TODO this is ugly. we should have a way to break out of the search early (here and in other methods).
+                    accessTimes = new TIntIntHashMap();
+                } else {
+                    accessTimes = sr.getReachedStops();
+                }
 
-            // getReachedStops returns distances, not times, so convert to times in the loop below
-            TIntIntMap accessTimes = sr.getReachedStops();
-            for (TIntIntIterator it = accessTimes.iterator(); it.hasNext(); ) {
-                it.advance();
-                // TODO how to handle kiss and ride/park and ride? Clearly this is not right.
-                it.setValue(it.value() / offstreetTravelSpeedMillimetersPerSecond);
+                // disallow non-transit access
+                // TODO should we allow non transit access with park and ride?
+                nonTransitTravelTimesToDestinations = new int[linkedDestinationsAccess.size()];
+                Arrays.fill(nonTransitTravelTimesToDestinations, RaptorWorker.UNREACHED);
```

Comment:
It seems that nonTransitTravelTimesToDestinations is filled with RaptorWorker.UNREACHED and then used in propagator where it is just read if P+R is used. In other modes it is filled with values based on states.